packages/strapi-plugin-upload/admin/src/components/BrowseAssets/index.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+
+import { PageFooter } from 'strapi-helper-plugin';
+
+import { generatePageFromStart, generateStartFromPage } from '../../containers/HomePage/utils';
+import Filters from '../Filters';
+import Flex from '../Flex';
+import List from '../List';
+import ListEmpty from '../ListEmpty';
+import Padded from '../Padded';
+import SelectAll from '../SelectAll';
+import SortPicker from '../SortPicker';
+import useModalContext from '../../hooks/useModalContext';
+import Wrapper from './Wrapper';
+
+const BrowseAssets = () => {
+  const {
+    count,
+    files,
+    goTo,
+    handleAllFilesSelection,
+    handleFileSelection,
+    multiple,
+    params,
+    removeFilter,
+    selectedFiles,
+    setParam,
+    sort,
+  } = useModalContext();
+
+  const handleChangeParams = ({ target: { name, value } }) => {
+    setParam({ name, value });
+  };
+
+  const handleChangeListParams = ({ target: { name, value } }) => {
+    if (name.includes('_page')) {
+      handleChangeParams({
+        target: { name: '_start', value: generateStartFromPage(value, params._limit) },
+      });
+    } else {
+      handleChangeParams({ target: { name: '_limit', value } });
+    }
+  };
+
+  const handleDeleteFilter = index => {
+    removeFilter(index);
+  };
+
+  const handleGoToUpload = () => {
+    goTo('browse');
+  };
+
+  const limit = parseInt(params._limit, 10) || 10;
+  const start = parseInt(params._start, 10) || 0;
+
+  const paginationParams = {
+    _limit: parseInt(params._limit, 10) || 10,
+    _page: generatePageFromStart(start, limit),
+  };
+
+  const areAllCheckboxesSelected = files.length === selectedFiles.length;
+  const hasSomeCheckboxSelected = selectedFiles.length > 0;
```

Comment:
Now we enable the multiple select through the pagination, the `areAllCheckboxesSelected` and `hasSomeCheckboxSelected` have to be adapted because I think we want the checkbox to be checked when all the items contained in a page are checked. As it we can select and unselect all items in the same page and keep other pages items selected. For now, we cannot unselect all items because the checkbox is never checked.

But maybe you wanted to do it in another PR!